jostedal/stun/agent.py

```diff
@@ -311,15 +305,15 @@ def __init__(self, data, family, port, address):
     @classmethod
     def decode(cls, data, offset, length):
         family, port = cls.struct.unpack_from(data, offset)
-        packed_ip = buffer(data, offset + cls.struct.size, length - cls.struct.size)
+        packed_ip = memoryview(data)[offset + cls.struct.size : offset + length]
```

Comment:
`offset + cls.struct.size : offset + cls.struct.size + length]`?

Reply:
buffer used start:length slice wants start:end so
length to end here is ```length - cls.struct.size``` to ```offset + cls.struct.size + length - cls.struct.size``` which (I think) simplifies to ```offset + length```

Reply:
Agreed. Addition is hard.